File: app/src/main/java/com/asfoundation/wallet/di/GasSettingsModule.java
package com.asfoundation.wallet.di;

import com.asfoundation.wallet.interact.FindDefaultNetworkInteract;
import com.asfoundation.wallet.viewmodel.GasSettingsViewModelFactory;
import dagger.Module;
import dagger.Provides;

@Module public class GasSettingsModule {

  @Provides public GasSettingsViewModelFactory provideGasSettingsViewModelFactory(
      FindDefaultNetworkInteract findDefaultNetworkInteract) {
    return new GasSettingsViewModelFactory(findDefaultNetworkInteract);
  }
}

File: app/src/main/java/com/asfoundation/wallet/di/GasSettingsModule.kt
package com.asfoundation.wallet.di

import com.asfoundation.wallet.home.usecases.FindNetworkInfoUseCase
import com.asfoundation.wallet.repository.GasPreferenceRepository
import com.asfoundation.wallet.ui.GasSettingsInteractor
import com.asfoundation.wallet.viewmodel.GasSettingsViewModelFactory
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@InstallIn(ActivityComponent::class)
@Module
class GasSettingsModule {
  @Provides
  fun provideGasSettingsViewModelFactory(gasSettingsInteractor: GasSettingsInteractor) =
      GasSettingsViewModelFactory(gasSettingsInteractor)

  @Provides
  fun providesGasSettingsInteractor(findNetworkInfoUseCase: FindNetworkInfoUseCase,
                                    gasPreferenceRepository: GasPreferenceRepository): GasSettingsInteractor {
    return GasSettingsInteractor(findNetworkInfoUseCase, gasPreferenceRepository)
  }
}